blueocean-core-js/src/js/jwt.js

```diff
@@ -0,0 +1,96 @@
+import fetch from 'isomorphic-fetch';
+import jwt from 'jsonwebtoken';
+import UrlUtils from './urlconfig';
+import { FetchFunctions } from './fetch';
+import { jwk2pem } from 'pem-jwk';
+let storedToken = null;
+let publicKeyStore = null;
+let tokenFetchPromise = null;
+export default {
+    /**
+     * Fetches the JWT token. This token is cached for a default of 25mins.
+     * If it is within 5mins or expiry it will fetch a new one.
+     */
+    fetchJWT() {
+        if (storedToken && storedToken.exp) {
+            const diff = storedToken.exp - Math.trunc(new Date().getTime() / 1000);
+            if (diff < 300) {
```

Comment:
So if, for some reason, the expiry on the token is set to less than 5 mins, will this not get into an infinite loop of requesting new tokens?